### PR TITLE
RefreshProgressIndicator to look native

### LINKED ISSUE #71086: RefreshProgressIndicator doesn't look like native
This issue relates to mismatches of `RefreshProgressIndicator` with swipe to refresh [material specs](https://material.io/design/platform-guidance/android-swipe-to-refresh.html) and native appearance

Related issue about `RefreshIndicator` #71075 

## Description
This is how `RefreshProgressIndicator` looks like now in flutter

* https://youtu.be/DWODhzmkSLU

Native swipe to refresh examples

* [Material spec](https://kstatic.googleusercontent.com/files/327b57850a9283ed6ff7af86d242a34626e959e71fcc991fbcc15b360b527329e2f37a06e3d5a233ab15e7d8a998d1d9879393f3b6e29aca8d0f663210222d98)
* FastHub https://youtu.be/cXjU_BWrM1U
* Google Chrome https://youtu.be/9J_PE3OqI-0
* GitHub https://youtu.be/TbltPdKOIWo

The arrow simply looks different and has different animation

Also, it's transperency at the start doesn't look good - arrow head is too small and you can see that the arrow actually consists of two parts

<div style="display:flex">

</div>

### PR DESCRIPTION
**Pull request by nt4f04uNd**
Fixes https://github.com/flutter/flutter/issues/71086

This makes the RefreshProgressIndicator to progress and look like the native Android's

Quick summary of changes:
- increased the stroke width
- changed the arrow shape
- increased the length the arrow head travels a little bit
- fixed arrow and arc overlapping each other with not full opacity creating the effect that can be seen in the issue
- increased the overall size by 1 pixel

Added a golden test for 'pulling' and 'indeterminate' states

<details>
<summary>Here's a comparison</summary>

| Flutter | Android |
| - | - |
| untitled | untitled |

</details>

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by bernaferrari** [comment]
Can someone test this in Android 12? I have no idea if they will change this 😛 maybe it wouldn't require a breaking change if it will change again in a few months.

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/88301

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/88301

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 4.
View them at https://flutter-gold.skia.org/cl/github/88301

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 5.
View them at https://flutter-gold.skia.org/cl/github/88301

**Comment by LasseRosenow** [comment]
Did you also fix the issue that the android version just moves back up a little on release while the flutter version moves up and also rotates back?

**Comment by nt4f04uNd** [comment]
@LasseRosenow not yet. This is related to RefreshIndicator (https://github.com/flutter/flutter/issues/71075), while the current PR only focuses on fixing the RefreshProgressIndicator (https://github.com/flutter/flutter/issues/71086)

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 7.
View them at https://flutter-gold.skia.org/cl/github/88301

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 8.
View them at https://flutter-gold.skia.org/cl/github/88301

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 9.
View them at https://flutter-gold.skia.org/cl/github/88301

File: packages/flutter/test/material/progress_indicator_test.dart [hunk 1]
```diff
@@ -859,3 +872,46 @@ void main() {
...
+    return Center(
+      child: Directionality(
+        textDirection: TextDirection.ltr,
+        child: RefreshProgressIndicator(
+          value: indeterminate ? null : controller.value,
+        ),
+      ),
+    );
+  }
+}
```
Comment (dkwingsmt) [inline packages/flutter/test/material/progress_indicator_test.dart:917]:
```suggestion
}

```

**Review by dkwingsmt — APPROVED** [review]
My apology that I missed the PR. It looks great! Can you update with master?

File: packages/flutter/test/material/progress_indicator_test.dart (hunk 1 above)
Comment (nt4f04uNd) [reply packages/flutter/test/material/progress_indicator_test.dart]: done

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 11.
View them at https://flutter-gold.skia.org/cl/github/88301

**Comment by Piinks** [comment]
The google testing shard and the flutter gold testing shard are stuck here. I am working to find out why, but in the meantime have confirmed they are passing and this is safe to land manually.

**Comment by nt4f04uNd** [comment]
> Can someone test this in Android 12? I have no idea if they will change this 😛 maybe it wouldn't require a breaking change if it will change again in a few months.

I just checked and it seems to not have changed

https://user-images.githubusercontent.com/39104740/136588204-6c8edfa5-5b76-41d6-b608-215b5d53ce5e.mp4

**Pull request merged**